core/src/main/java/hudson/model/ParameterDefinition.java

```diff
@@ -103,11 +104,11 @@
 
     private final String description;
 
-    public ParameterDefinition(String name) {
+    public ParameterDefinition(@Nonnull String name) {
```

Comment:
Making this bit nonnull makes sense to me because there seems to be no way to obtain that parameter if this is null. Although i don't know how pipeline can enforce this

Reply:
Seems to make sense.